src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -691,16 +691,29 @@ def auto_transfer(self, service_id: uuid.UUID) -> inmanta_lsm.model.ServiceInsta
             transfer_type=inmanta_lsm.const.TransferTrigger.AUTO,
         )
 
-        def next_state(state: str) -> None:
+        def next_state(state: str, is_error_transition: bool = False) -> None:
             """
             Apply this state to our service, if it is different from the current
             state, also increment the version
 
             :param state: The new state to apply
+            :param is_error_transition: Is it an error transition?
             """
             if service.state == state:
                 return
 
+            try:
+                if is_error_transition:
+                    is_preserving_same_desired_state = transfer.error_same_desired_state
+                else:
+                    is_preserving_same_desired_state = transfer.target_same_desired_state
+
+                if not is_preserving_same_desired_state:
+                    service.desired_state_version += 1
```

Comment:
Yes, but:
1. I would rather do it at the same place we bump the service version
2. We also need to skip the compile if we don't emit a new desired state version

Reply:
I realize now that 2. is not that easy, because in this case, the compile that was called in the method is the one on the current state, not the one triggered after the state transfer.